src/Faction/ui/FactionsRoot.tsx

```diff
@@ -79,6 +79,9 @@ export function FactionsRoot(props: IProps): React.ReactElement {
       (augmentation: string) => !player.hasAugmentation(augmentation)
     ).length;
   }
+  let allJoinedFactions = props.player.factions
+  allJoinedFactions = allJoinedFactions.sort((a, b) =>
+    allJoinedFactions.indexOf(a) - allJoinedFactions.indexOf(b));
```

Comment:
the sorting works in-place: see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
so the use of the return value is unnecessary.
it might also be good to copy the array instead of using `props.player.factions` directly

did you want to use `allFactionsThatExist.indexOf()` ?

Reply:
Is it intended that you change the `props.player.factions` array directly from UI code instead of working with a copy?

Reply:
hmm but by calling sort dont we get a new copy returned?